### PR TITLE
Windows: improve Win32 API compatibility

### PR DESCRIPTION
**Pull request by CarterLi**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -76,6 +77,15 @@ if(NOT CMAKE_BUILD_TYPE)
 endif()
 
 message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+if(ENABLE_THREADS)
+    if(CMAKE_USE_PTHREADS_INIT)
+        message(STATUS "Threads type: pthread")
+    else()
+        message(STATUS "Threads type: Win32 thread")
+    endif()
```
Comment (LinusDierheimer) [inline CMakeLists.txt:85]:
```cmake
message(STATUS "Threads type: ${CMAKE_THREAD_LIBS_INIT}")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -482,6 +492,14 @@ ff_lib_enable(OPENCL OpenCL)
 ff_lib_enable(LIBCJSON libcjson)
 ff_lib_enable(FREETYPE freetype2)
 
+if(ENABLE_THREADS)
+    target_compile_definitions(libfastfetch PRIVATE FF_HAVE_THREADS)
+
+    if(CMAKE_USE_PTHREADS_INIT)
+        target_link_libraries(libfastfetch PRIVATE Threads::Threads)
+    endif()
```
Comment (LinusDierheimer) [inline CMakeLists.txt:500]: I don't think we need the if, `Threads::Threads` will be empty if threads is provided by the standard lib.

**Comment by LinusDierheimer** [comment]
I like the abstraction away from pthreads!

File: CMakeLists.txt (hunk 1 above)
Comment (CarterLi) [reply CMakeLists.txt:85]: Just tested it. CMAKE_THREAD_LIBS_INIT was empty on my mac

Comment (LinusDierheimer) [reply CMakeLists.txt:85]: Ah that may be because pthread is part of libc there.

Comment (CarterLi) [reply CMakeLists.txt:85]: Let's keep it as it is.

**Pull request merged**